### PR TITLE
Metrics in e2e tests are using printer framework to print themselves

### PR DESCRIPTION
**Pull request by gmarek**
cc @wojtek-t

### CODE REVIEW

File: pkg/metrics/api_server_metrics.go [hunk 1]
```diff
@@ -48,6 +48,24 @@ var KnownApiServerMetrics = map[string][]string{
 	"rest_client_request_status_codes":               {"code", "host", "method"},
 }
 
+var InterestingApiServerMetricsForE2E = []string{
```
Comment (fgrzadkowski) [inline pkg/metrics/api_server_metrics.go]: I guess that this [comment](https://github.com/kubernetes/kubernetes/pull/19261/files#r48809561) also applies here. Do you plan to change it?

**Comment by gmarek** [comment]
I'm changing this PR after comments from @timothysc

File: test/e2e/framework.go [hunk 2]
```diff
@@ -161,6 +161,21 @@ func (f *Framework) afterEach() {
...
+		// TODO: enable Scheduler and ControllerManager metrics grabbing when Master's Kubelet will be registered.
+		grabber, err := metrics.NewMetricsGrabber(f.Client, true, false, false, true)
+		if err != nil {
+			Logf("Failed to create MetricsGrabber. Skipping metrics gathering.")
+		} else {
+			received, err := grabber.Grab(nil)
+			if err != nil {
+				Logf("MetricsGrabber failed grab metrics. Skipping metrics gathering.")
+			} else {
+				summaries = append(summaries, (*metricsForE2E)(&received))
```
Comment (fgrzadkowski) [inline test/e2e/framework.go]: I don't understand this line. You're casting `received` to `metricsForE2E`. But shouldn't you also call something like `PrintHumanReadable` on it?

Comment (gmarek) [inline test/e2e/framework.go]: Yup. received has a type MetricsCollection which does not have PrintHumanReadable method, and metricsForE2E implements Summary interface (i.e. implement PrintHumanReadable and PrintJSON). It's a simple 'wrapper' type for MetricsCollection that adds those methods to them.

**Comment by mwielgus** [comment]
LGTM

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Comment by gmarek** [comment]
@k8s-bot unit test this

**Pull request merged**